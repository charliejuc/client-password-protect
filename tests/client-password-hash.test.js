const clientPasswordProtect = require('../')

test('clientPasswordProtect should be a function', () => {
    expect(clientPasswordProtect)
        .toBeInstanceOf(Function)
})

test('Hash match snapshot', () => {
    expect(clientPasswordProtect('Hello'))
        .toMatchSnapshot()

    expect(clientPasswordProtect('dsaFdsfsa97 //sas 2342sadfsa dasfsa**'))
        .toMatchSnapshot()
})

test('Hash mismatch when change rounds', () => {
    expect(clientPasswordProtect('Hello'))
        .not
        .toBe(clientPasswordProtect('Hello', { rounds: 100 }))
})

test('Called with no params', () => {
    expect(() => clientPasswordProtect())
        .toThrow("Password can't be empty")
})

test('Password as empty string', () => {
    expect(() => clientPasswordProtect(''))
        .toThrow("Password can't be empty")
})

test('Rounds option as null', () => {
    expect(() => clientPasswordProtect('Hello', { rounds: null }))
        .toThrow('should be a positive integer')
})

test('Options as empty object({})', () => {
    expect(() => clientPasswordProtect('Hello', {}))
        .not
        .toThrow()

    expect(clientPasswordProtect('Hello', {}))
        .toBe(clientPasswordProtect('Hello'))
})

test('Rounds option as undefined', () => {
    expect(() => clientPasswordProtect('Hello', { rounds: undefined }))
        .not
        .toThrow()

    expect(clientPasswordProtect('Hello', { rounds: undefined }))
        .toBe(clientPasswordProtect('Hello'))
})