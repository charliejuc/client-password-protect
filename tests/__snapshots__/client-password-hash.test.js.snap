// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hash match snapshot 1`] = `"e37489641838fabc8a516f349c5d64c5a1cdb31998fc0800b8e2de326a7b4a9d"`;

exports[`Hash match snapshot 2`] = `"3e6c7cc437327b0b613d30a916834884606f9d2ad7157925e0bd572a145e6ee6"`;
